packages/web-components/fast-foundation/src/menu-item/menu-item.template.ts

```diff
@@ -9,18 +9,42 @@ import { MenuItem, MenuItemRole } from "./menu-item";
 export const MenuItemTemplate = html<MenuItem>`
     <template
         role="${x => x.role}"
+        aria-haspopup="${x =>
+            x.submenuElements && x.submenuElements.length > 0 ? "menu" : void 0}"
         aria-checked="${x => (x.role !== MenuItemRole.menuitem ? x.checked : void 0)}"
         aria-disabled="${x => x.disabled}"
         aria-expanded="${x => x.expanded}"
         @keydown="${(x, c) => x.handleMenuItemKeyDown(c.event as KeyboardEvent)}"
         @click="${(x, c) => x.handleMenuItemClick(c.event as MouseEvent)}"
         class="${x => (x.disabled ? "disabled" : "")} ${x =>
             x.expanded ? "expanded" : ""}"
+        ${children({
+            property: "submenuElements",
+            filter: elements("[slot=submenu]"),
+        })}
     >
         ${startTemplate}
         <span class="content" part="content">
             <slot></slot>
         </span>
         ${endTemplate}
+        ${when(
+            x => x.expanded,
+            html<MenuItem>`
+                <fast-anchored-region
+                    :anchorElement="${x => x}"
+                    vertical-positioning-mode="dynamic"
+                    vertical-default-position="bottom"
+                    vertical-inset="true"
+                    horizontal-positioning-mode="dynamic"
+                    horizontal-default-position="end"
+                    class="submenu-region"
+                    dir="${x => x.currentDirection}"
+                    ${ref("subMenuRegion")}
+                >
```

Comment:
I think we're going to need to fundamentally change the DOM here for the sake of submenu items. A simple example of what submenu DOM structure needs to look like to be semantically accurate and provide the right hierarchy for Assistive Technology.

```html
<button 
    role="menuitem"
    aria-haspopup="true"
    aria-expanded="false"
>
    The menu item content
</button>
<ul 
    role="menu" 
    aria-label="Facts"
>
    <li role="none">
        <a role="menuitem" href="#">
        History
        </a>
    </li>
    <li role="none">
        <a role="menuitem" href="#">
        Current Statistics
        </a>
    </li>
    <li role="none">
        <a role="menuitem" href="#">
        Awards
        </a>
    </li>
</ul>
```

Reply:
So the example structure here isn't valid?  https://www.w3.org/WAI/GL/wiki/Using_ARIA_menus

Which is roughly:

```html
<ul role="menubar">
  <li role="menuitem">
    Font
    <ul role="menu">
      <li role="menuitemradio">Sans-serif</li>
      <li role="menuitemradio">Serif</li>
    </ul>
  </li> r
  <li role="menuitem">
    Style
    <ul role="menu">
      <li role="menuitemradio">Sans-serif</li>
      <li role="menuitemradio">Serif</li> 
    </ul>
</ul>
```
Which is essentially the structure we create at runtime:
![image](https://user-images.githubusercontent.com/7649425/101539353-405f7a80-3953-11eb-9c11-d7a8e3cf021f.png)